### ISSUE TITLE
Is there a way to cut sessions after recording ?

### ISSUE THREAD

**Issue by lubenard** [rrweb-io/rrweb#398]
Hello,
I am trying to cut sessions after recording.

For example, if i have a 1 mn 30s record, but just want to keep from 10 sec to 30 sec and 1mn to 1mn 10 sec.

The events are stored into a array, so i can cut them as i want. 

The 1 and 2 array elements contains the web page, and all of the following events are just key / mouse events if i understand it well.

Is there a way to record the full dom into each event ?
I am missing something ?

Thanks !

**Comment by Yuyz0112** [comment]
This is a good feature request, like editor mode for the session.

Currently, the data structure of the events is an operation log so we could not cut part of them.

**Comment by lubenard** [comment]
Yes. Couldn't we record full dom each time (as an option obviously) so that we could just cut the array and still get all the modification ?

If you have entry point in the code, i'll be happy to have a look !

**Comment by Yuyz0112** [comment]
Does [checkout feature](https://github.com/rrweb-io/rrweb/blob/master/guide.md#checkout) works for you?

**Comment by lubenard** [comment]
Actually not, because i am building a rrweb record editor.
It will look like a basic video editor, with a timeline and parts of the record you could delete.

**Comment by dwootton** [comment]
+1 to this feature although I don't think we'd need to do a full record at each event (which could be prohibitively costly). In this case, what we really need is to be able to do "recreate" a checkout at any given index. For @lubenard's case, when you go to trim the start time, the user would select the index they want to trim to and then we would need to use the logged events to recreate a checkout at that time. I'm still ramping up on the internals of the library @Yuyz0112, but do you think this would be possible?

**Comment by lubenard** [comment]
@dwootton Your idea is really good !

We could create a method like 'replayer.cut(seconds)' or 'replayer.cut(arrayIndex)'.

It would apply all events created from the index 0 to index N (choosen) or second choosen.

**Comment by eoghanmurray** [comment]
You need to collect the most recent `full-snapshot` event before the start of the first event after the cut right?
And also any DOM mutation events that would otherwise have been cut.
Then apply those immediately after the cut period.
